app/controllers/users/messages_controller.rb

```diff
@@ -27,6 +27,7 @@ def contact
 
   def set_recipient
     @recipient_user = User.find_by(url_name: params[:url_name])
+    raise ActiveRecord::RecordNotFound unless @recipient_user
```

Comment:
https://api.rubyonrails.org/v4.2.10/classes/ActiveRecord/FinderMethods.html#method-i-find_by-21 is a better choice for doing this.